python/lsst/ap/association/diaPipe.py

```diff
@@ -97,17 +97,31 @@ class DiaPipelineConnections(
     )
     associatedDiaSources = connTypes.Output(
         doc="Optional output storing the DiaSource catalog after matching, "
-            "calibration, and standardization for insertation into the Apdb.",
+            "calibration, and standardization for insertion into the Apdb.",
         name="{fakesType}{coaddName}Diff_assocDiaSrc",
         storageClass="DataFrame",
         dimensions=("instrument", "visit", "detector"),
     )
+    diaForcedSources = connTypes.Output(
```

Comment:
Is the flags column always a float, or is it only a float after you use the pandas `concat` function?

There is no way that I know of to enforce a schema on pandas tables (hence why many of us want to push the project away from using them), so I don't think what you're asking for is possible either way.

Reply:
The flags column is always a float, when it exists; it does not exist for the first visit+detector processed, but it does for the second, and it is a float then.

Reply:
`diaForcedSources['flags']` is added when `self.diaCatalogLoader` returns `>1` entries (shouldn't that be `>0`?) in `loaderResult.diaForcedSources` (thus, only when there are previous sources in this field to do forced photometry on, I believe). The `append` call in `doPackageAlerts` results in that field being converted to `float64` from the originally-loaded `int64`: that's definitely a bug, and @kfindeisen 's changes on #162 won't fix it, either. This is, once again, the danger of using pandas.

We could conceivably have `diaForcedSource.run` always add a zero-filled `flags` column, so that pandas doesn't try to use it's terribly stupid "null means convert numbers to float" approach.

I don't know what goes into the `DiaForceSource['flags']` field, and apparently neither does the APDB baseline schema in `sdm_schemas`, since the docstring for all our `flags` columns are: "bitfield, tbd", so I don't know how necessary this particular field is in this particular schema, or what it even means.